src/Microsoft.Data.SqlClient/src/Microsoft/Data/SqlClient/Server/SqlNormalizer.cs

```diff
@@ -20,23 +20,23 @@ internal sealed class FieldInfoEx : IComparable
 
         internal FieldInfoEx(FieldInfo fi, int offset, Normalizer normalizer)
         {
+            _offset = offset;
+            FieldInfo = fi;
             Debug.Assert(normalizer != null, "normalizer argument should not be null!");
             Normalizer = normalizer;
-            FieldInfo = fi;
-            _offset = offset;
         }
+
         internal FieldInfo FieldInfo { get; private set; }
         internal Normalizer Normalizer { get; private set; }
 
         // Sort fields by field offsets.
         public int CompareTo(object other)
         {
-            FieldInfoEx otherF = other as FieldInfoEx;
-            if (otherF == null)
+            if (other is FieldInfoEx otherFieldInfoEx)
             {
-                return -1;
+                return _offset.CompareTo(otherFieldInfoEx._offset);
             }
-            return _offset.CompareTo(otherF._offset);
+            return -1;           
         }
```

Comment:
Just as a suggestion: using `is` pattern.
```
public int CompareTo(object other) 
     => other is FieldInfoEx otherF ? _offset.CompareTo(otherF._offset): -1;
```

Reply:
Done.